nameserver/server_test.go

```diff
@@ -112,6 +116,76 @@ func TestDNSServer(t *testing.T) {
 	// needs two servers, each listening on its own address
 }
 
+func TestTCPDNSServer(t *testing.T) {
+	const (
+		port          = 17625
+		localTestName = "test1.weave.local."
+		testAddr1     = "10.0.2.1"
+	)
+	dnsAddr := fmt.Sprintf("localhost:%d", port)
+
+	InitDefaultLogging(true)
+	var zone = new(ZoneDb)
+
+	// Insert a bunch of IPs for the same name
+	ansCount := 0
+	for i := 0; i < 255; i++ {
+		for j := 0; j < 10; j++ {
+			ip := net.ParseIP(fmt.Sprintf("10.0.%d.%d", i, j))
+			ansCount += 1
+			err := zone.AddRecord(containerID, localTestName, ip)
+			wt.AssertNoErr(t, err)
+		}
+	}
+
+	// Run another DNS server for fallback
+	s, fallbackAddr, err := RunLocalUDPServer("127.0.0.1:0")
+	wt.AssertNoErr(t, err)
+	defer s.Shutdown()
+
+	_, fallbackPort, err := net.SplitHostPort(fallbackAddr)
+	wt.AssertNoErr(t, err)
+
+	config := &dns.ClientConfig{Servers: []string{"127.0.0.1"}, Port: fallbackPort}
+	srv, err := NewDNSServerWithConfig(config, zone, nil, port)
+	wt.AssertNoErr(t, err)
+	defer srv.Stop()
+	go srv.Start()
```

Comment:
I don't understand why this is a goroutine. `Start` does itself spawn goroutines, and, afaict, does terminate.